src/contract.ts

```diff
@@ -58,21 +64,67 @@ export class AbstractContract {
     this.scriptedConstructor.init(asmVarValues);
   }
 
+  static findMappableAsmWord(debugAsm: DebugModeAsmWord[], pc: number): DebugModeAsmWord | undefined {
+    while (--pc > 0) {
+      if (debugAsm[pc].file && debugAsm[pc].file !== "std" && debugAsm[pc].line > 0) {
+        return debugAsm[pc];
+      }
+    }
+  }
+
+
   run_verify(unlockingScriptASM: string, txContext?: TxContext): VerifyResult {
     const txCtx: TxContext = Object.assign({}, this._txContext || {}, txContext || {});
 
-    const us = bsv.Script.fromASM(unlockingScriptASM);
+    const us = bsv.Script.fromASM(unlockingScriptASM.trim());
     const ls = this.lockingScript;
     const tx = txCtx.tx;
     const inputIndex = txCtx.inputIndex || 0;
     const inputSatoshis = txCtx.inputSatoshis || 0;
 
-    const si = bsv.Script.Interpreter();
-    const result = si.verify(us, ls, tx, inputIndex, DEFAULT_FLAGS, new bsv.crypto.BN(inputSatoshis));
+    const bsi = bsv.Script.Interpreter();
+  
+    let stepCouter = 0;
+		bsi.stepListener = function (step: any, stack: any[], altstack: any[]) {
+      stepCouter++;
+    };
+
+    
+    const debugAsm: DebugModeAsmWord[] =  Object.getPrototypeOf(this).constructor.debugAsm;
+
+    const result = bsi.verify(us, ls, tx, inputIndex, DEFAULT_FLAGS, new bsv.crypto.BN(inputSatoshis));
+
+    let error = bsi.errstr;
+
+    const offset = unlockingScriptASM.trim().split(' ').length;
+    
+    // the complete script will have op_return and data, but debugAsm do not have, so we need to make sure the index in bound.
```

Comment:
// the complete script may have op_return and data, but compiled output does not have it. So we need to make sure the index is in boundary.

Reply:
fixed